orchagent/portsorch.cpp

```diff
@@ -6414,6 +6599,12 @@ void PortsOrch::createPortBufferQueueCounters(const Port &port, string queues)
             /* add watermark queue counters */
             addQueueWatermarkFlexCountersPerPortPerQueueIndex(port, queueIndex);
         }
+
```

Comment:
The function `createPortBufferQueueCounters` is called for each queue whose `BUFFER_QUEUE` is configured. Originally, the function was introduced to optimize queue counter polling. The idea is that only queue counters need to be polled on queues that are really used by users, which is indicated by whether the buffer is configured.
The optimization is done just because counter-polling is not for free but consumes a lot of CPU resources.
I think there can be a stronger condition for the WRED counter: only queues configured with WRED profiles need to be polled.
Can we do that?

Reply:
1.	Wred stats are disabled by default. The user needs to enable it explicitly using the counterpoll command (“counterpoll wredqueue enable”). And at syncd, the existing queue flexcounter manager polling the queue stats will include the additional WRED/ECN stats. No new threads are added at this level. Hence, this will not impact the current users unless they explicitly enable WRED stats.
2.	Even when WRED stats is enabled, the existing queue flexcounter manager will include the WRED stat-ids along with the other queue counters it is currently polling. It is not adding additional SAI calls.
 
The proposed optimization is not helpful in the common case: It is not common to have WRED profile enabled but not the buffer-profile. Similarly it is uncommon to have only WRED queue counters but not the other queue counters. Also, we are very close to the forking of the 202305 release branch. I will consider this suggestion in the future release.

Reply:
I'm ok if the optimization will not be implemented in the current release but suggest mentioning it in the HLD as a possible optimization.
It can take extra time to fetch additional counters even if it doesn't increase the number of SAI API calls.